app/src/main/java/blockeq/com/stellarwallet/utils/AccountUtils.kt

```diff
@@ -37,10 +44,25 @@ class AccountUtils {
                 cipherWrapper.encrypt(mnemonic, masterKey?.public)
             } else {
                 WalletApplication.localStore.isPassphraseUsed = true
+                WalletApplication.localStore.encryptedPassphrase = cipherWrapper.encrypt(passphrase!!, masterKey?.public)
                 cipherWrapper.encrypt("$mnemonic $passphrase", masterKey?.public)
             }
         }
 
+        fun getDecryptedPhrase(encryptedPhrase: String, masterKey: java.security.KeyPair) : String {
+            val cipherWrapper = CipherWrapper(CIPHER_TRANSFORMATION)
+            return cipherWrapper.decrypt(encryptedPhrase, masterKey.private)
+        }
+
+        fun getDecryptedPassphhrase(encryptedPassphrase: String?, masterKey: java.security.KeyPair) : String? {
+            return if (WalletApplication.localStore.isPassphraseUsed && encryptedPassphrase != null) {
```

Comment:
should the flag `isPassphraseUsed` be removed since the logic could relay on the existence of the the `encryptedPassphrase`?

Reply:
I think is clear if we do something like
```
isOldWalletWithPassphrase(){
 return if (WalletApplication.localStore.isPassphraseUsed && encryptedPassphrase != null) 
}
```